src/net_processing.cpp

```diff
@@ -1475,6 +1476,11 @@ void PeerManagerImpl::FindNextBlocks(std::vector<const CBlockIndex*>& vBlocks, c
                 return;
             }
 
+            // Don't request blocks that go further than what limited peers can provide
+            if (is_limited_peer && (state->pindexBestKnownBlock->nHeight - pindex->nHeight >= static_cast<int>(NODE_NETWORK_LIMITED_MIN_BLOCKS) - 2 /* two blocks buffer for possible races */)) {
```

Comment:
I'm not sure what you are referring to. Could you rephrase it please.
You want to update `pindexLastCommonBlock` to `pindex`?

Reply:
I think what @naumenkogs is referring to here is that if this condition is true then `pindex` will hit this condition on the first `pindex` in the loop and every `pindex` up until this condition is no longer true. This can be more efficient by just bumping `pindex` to the first `pindex` that is past the network limited blocks threshold.

So, we can either change the loop from range based to regular for-loop and advance the index, or move this check out of this loop and into the loop on line 1464 and break early if the condition is met.

Reply:
> I think what @naumenkogs is referring to here is that if this condition is true then `pindex` will hit this condition on the first `pindex` in the loop and every `pindex` up until this condition is no longer true. This can be more efficient by just bumping `pindex` to the first `pindex` that is past the network limited blocks threshold.
>
> So, we can either change the loop from range based to regular for-loop and advance the index, or move this check out of this loop and into the loop on line 1464 and break early if the condition is met.

Ok, we are on the same page now. Thanks. Let's leave it for a follow-up. It would be nice to re-think this function and benchmark it properly. In a first glance, I think we would be optimizing the function for the less likely case and may degrade performance for the likely ones.